### PR TITLE
Bug po box vector

### PR DESCRIPTION
**Pull request by mixmix**
I found a bug!

this test vector for po-box encryption was not a the right BFE type.

What I've done:
- wrote a test to catch incorrect BFE type-format for keys
- fixed the test vector
    - generated the test vector output using `ssb-pirvate-group-keys` module code

### CODE REVIEW

File: vectors/po-box-key1.json [hunk 1]
```diff
@@ -5,11 +5,11 @@
     "my_dh_secret": "AwBg/xkWz7qNmtIz0jBWVgiFQoHIFqbJKP9JpYSx1VVbVQ==",
     "my_dh_public": "AwDZg4xIRpi3tbZsfuWVRnbYHNDo33pEBQIiQZaC2TZuPg==",
     "my_feed_id": "AADXDOY5Isl/kWTPOYAryv6lTcDlCQnHb8a35mDCCSILSQ==",
-    "po_box_dh_public": "AwAIFlU+iztEMB7eLSP6fZtpqdt5DFemgSmvqaGbx8I4ZQ==",
```
Comment (mixmix) [inline vectors/po-box-key1.json:8]:
```
<Buffer 03 00 08 16 55 3e 8b 3b 44 30 1e de 2d 23 fa 7d 9b 69 a9 db 79 0c 57 a6 81 29 af a9 a1 9b c7 c2 38 65>
```

[3, 0] = BFE code for DM encryption key

File: vectors/po-box-key1.json [hunk 2]
```diff
@@ -5,11 +5,11 @@
...
+    "po_box_dh_public": "AwEIFlU+iztEMB7eLSP6fZtpqdt5DFemgSmvqaGbx8I4ZQ==",
```
Comment (mixmix) [inline vectors/po-box-key1.json:8]:
```
<Buffer 03 01 08 16 55 3e 8b 3b 44 30 1e de 2d 23 fa 7d 9b 69 a9 db 79 0c 57 a6 81 29 af a9 a1 9b c7 c2 38 65>
```
[3, 1] = BFE code for PO Box encryption key

**Comment by mixmix** [comment]
Recommend this is a *Semver Patch version* release

File: test.js [hunk 3]
```diff
@@ -12,6 +19,19 @@ fs.readdir(path.join(__dirname, 'vectors'), (_, fileNames) => {
...
       isTrue(isObject(vector.input), 'input')
+
+      for (const key in vector.input) {
+        if (
+          key === 'my_dh_secret' ||
+          key === 'my_dh_public' ||
+          key === 'your_dh_public'
+        ) isTrue(isTypeFormat(vector.input[key], TF.encryptionKeys.dm), key + ' is dm key')
+
+        if (
```
Comment (arj03) [inline test.js:30]: I would prefer an `else if` here

Comment (mixmix) [reply test.js:30]: It's a test, but yes sure can change

**Comment by arj03** [comment]
Funny coincidence you are opening a fix against this, I was looking at this spec and BFE spec recently and was wondering where exactly these "3 encryption formats" are used. Can you help me better understand that?

**Comment by mixmix** [comment]
They're used in box2 dms / pobox. Keys that go into deriving a shared key are bfe encoded for the "info" part of some of it
Keep in mind bwe don't use ed25519 keys, we use curve25519.
So we choose which sort of key it is dm or pobox.

Can give you more specificity when at computer if needed. Or look in ssb-private-group-keys to see dm / pobox stuff.

I'm about to open a pr there to fix/ extend some things I need for ssb-keys. Also found a gotcha over there...

File: test.js (hunk 3 above)
Comment (mixmix) [reply test.js]: Done (force pushed over)

**Pull request merged**